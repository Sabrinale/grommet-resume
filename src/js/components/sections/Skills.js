import React from 'react'
import { Box } from 'grommet'

import { SecondaryTitle } from '../SecondaryTitle'
import { SectionTitle } from '../SectionTitle'
import { Highlights } from '../Highlights'
import { TagSection } from '../TagSection'

const Skill = ({ item }) => (
  <Highlights>
    <SectionTitle>{item.name}</SectionTitle>
    <Box>
      <SecondaryTitle>{`Level: ${item.level}`}</SecondaryTitle>
      <TagSection tags={item.keywords}>Keywords</TagSection>
    </Box>
  </Highlights>
)

export { Skill }
