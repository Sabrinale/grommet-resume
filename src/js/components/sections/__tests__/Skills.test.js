import React from 'react'
import renderer from 'react-test-renderer'

import { Skill } from '../Skills'

import defaultResume from './mockResume.json'

test('renders skills section', () => {
  const data = defaultResume.skills[0]
  const component = renderer.create(<Skill item={data} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
