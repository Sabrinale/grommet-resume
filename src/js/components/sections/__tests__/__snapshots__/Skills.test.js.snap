// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders skills section 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 kFdcfG"
>
  <h3
    className="StyledHeading-sc-1rdh4aw-0 jVEgcZ"
  >
    Web Development
  </h3>
  <div
    className="StyledBox-sc-13pk1d4-0 IRSNj"
  >
    <h4
      className="StyledHeading-sc-1rdh4aw-0 hlZfma"
    >
      Level: Master
    </h4>
    <div
      className="StyledBox-sc-13pk1d4-0 IRSNj"
    >
      <span
        className="StyledText-sc-1sadyjn-0 dndifk"
        size="medium"
      >
        Keywords
      </span>
      <div
        className="StyledBox-sc-13pk1d4-0 dvQWRk"
      >
        <span
          className="StyledText-sc-1sadyjn-0 fWSbXS"
        >
          HTML
        </span>
        <span
          className="StyledText-sc-1sadyjn-0 fWSbXS"
        >
          CSS
        </span>
        <span
          className="StyledText-sc-1sadyjn-0 fWSbXS"
        >
          Javascript
        </span>
      </div>
    </div>
  </div>
</div>
`;
